ProjWidget.py

```diff
@@ -5,6 +5,111 @@
 import param
 from matplotlib.backends.backend_agg import FigureCanvasAgg as FigureCanvas
 from matplotlib.figure import Figure
+from lmfit.models import GaussianModel, Model, update_param_vals
+import numpy as np
+
+# From lmfit.lineshapes, because it's not worth importing...
+#
+# tiny had been numpy.finfo(numpy.float64).eps ~=2.2e16.
+# here, we explicitly set it to 1.e-15 == numpy.finfo(numpy.float64).resolution
+tiny = 1.0e-15
+
+
+def sg4(x, amplitude=1.0, center=0.0, width=1.0):
+    return amplitude * np.exp(-2.0 * ((x - center) ** 4 / max(tiny, width**4)))
+
+
+def sg6(x, amplitude=1.0, center=0.0, width=1.0):
+    return amplitude * np.exp(-2.0 * ((x - center) ** 6 / max(tiny, width**6)))
+
+
+class SG4Model(Model):
```

Comment:
Where'd these come from? Copied/modified from lmfit's models?

Reply:
Yes.  lmfit had a gaussian model that evidently played games to avoid problems with small numbers.  I thought these guys probably knew what they were doing, so I basically copied their gaussian model as a template and changed the computation to an sg4 or sg6 model.